<h1>Table of Contents<span class="tocSkip"></span></h1>
<div class="toc"><ul class="toc-item"><li><span><a href="#Тема-1.-Введение-в-программирование-на-языке-Python" data-toc-modified-id="Тема-1.-Введение-в-программирование-на-языке-Python-1"><span class="toc-item-num">1&nbsp;&nbsp;</span>Тема 1. Введение в программирование на языке Python</a></span></li><li><span><a href="#Задания" data-toc-modified-id="Задания-2"><span class="toc-item-num">2&nbsp;&nbsp;</span>Задания</a></span></li><li><span><a href="#1.-Числовые-и-строковые-типы-данных-языка-Python.-Управляющие-конструкции." data-toc-modified-id="1.-Числовые-и-строковые-типы-данных-языка-Python.-Управляющие-конструкции.-3"><span class="toc-item-num">3&nbsp;&nbsp;</span>1. Числовые и строковые типы данных языка Python. Управляющие конструкции.</a></span><ul class="toc-item"><li><span><a href="#1.1.-Работа-с-числами.-Базовые-числовые-типы-int-и-float" data-toc-modified-id="1.1.-Работа-с-числами.-Базовые-числовые-типы-int-и-float-3.1"><span class="toc-item-num">3.1&nbsp;&nbsp;</span>1.1. Работа с числами. Базовые числовые типы int и float</a></span></li><li><span><a href="#1.2.-Строки,-функции-и-методы-работы-со-строками" data-toc-modified-id="1.2.-Строки,-функции-и-методы-работы-со-строками-3.2"><span class="toc-item-num">3.2&nbsp;&nbsp;</span>1.2. Строки, функции и методы работы со строками</a></span></li></ul></li></ul></div>


# Тема 1. Введение в программирование на языке Python

# Задания

Выполните задания

# 1. Числовые и строковые типы данных языка Python. Управляющие конструкции.

## 1.1. Работа с числами. Базовые числовые типы int и float

1. Значения переменных А и В ввести с клавиатуры и вывести на экран. После этого значения меняются местами, т.е. A нужно присвоить значение B, а B – значение A, и вновь значения переменных вывести на экран.

2. Значение x вводится с клавиатуры. Вычислите $y=(x+1)^2\cdot \sqrt{x^3+1}$ Выведите на экран значения $x$ и $y$ с тремя знаками после запятой.

3. Вычислите сумму цифр пятизначного числа.

4. Для заданного трехзначного числа выведите число, у которого цифры идут в обратном порядке, например, для числа 123 ответ 321.

5. Ввести координаты 2 точек: $(x_1,y_1)$ и $(x_2,y_2)$. Вычислите расстояние между этими точками.  Результат выведите с 5 знаками после запятой.

6. Разработать программу вычисления по известному радиусу площади круга и длины окружности.

7. Разработать программу по вычислению площади кольца по известным значениям его внешнего и внутреннего радиусов.

8. Разработать программу вычисления радиуса круга и его площади по известной длине окружности.

9. Разработать программу вычисления объема шара: а) по радиусу; б) по длине его экваториальной параллели. Объём шара рассчитывается по формуле $V=\frac{4}{3}\pi{r^3}$, где $r$ - радиус.

10. Имеются два $n$–мерных вектора $x$ и $y$, которые задают координаты $n$ точек на плоскости (случайные целые числа). Найти наиболее близкие друг другу точки.

11. Треугольник задан координатами своих вершин: $(x_1,y_1),(x_2,y_2),(x_3,y_3)$. Значения координат определите с помощью присваивания. Они могут быть нецелыми. Найти периметр и площадь треугольника.

12. Пусть пользователь вводит 5 чисел: а, b, c, d, e. Реализуйте программу расчета выражения вида $res=\frac{a+bc}{2d-e}+Ost[b^e/c]$, где $Ost$ - остаток от деления. Учесть невозможность деления на 0.


## 1.2. Строки, функции и методы работы со строками

1. Введите строку, состоящую из двух цифр. Преобразуйте ее в целое и вещественное число. Выведите полученные три значения (строку, целое число, вещественное число) на экран в одной строке через запятую, затем пропустите строку и вновь выведите значения по одному на строке. Перед каждым значением выведите его тип.

2. Для строки 'Финансовый университет' двумя способами получить подстроку с 1-го по 4-й символы включительно.

3. Составить строку из всех четных символов строки 'Финансовый университет'.

4. Получить подстроку неизвестной заранее строки, содержащую половину символов строки и расположенную посередине строки.

5. Коротко записать создание строки 'oneoneoneoneoneonetwotwotwo'

6. Для введенной строки выведите (на отдельной строке):
- второй символ этой строки;
- предпоследний символ этой строки;
- всю строку, кроме последних двух символов;
- все символы с четными индексами (считая, что индексация начинается с 0, поэтому символы выводятся, начиная с первого);
- все символы с нечетными индексами, то есть начиная со второго символа строки;
- все символы в обратном порядке;
- все символы строки через один в обратном порядке, начиная с последнего;
- длину данной строки.

7. Не используя метод count, для заданной строки выполните:
- если символ * в данной строке отсутствует, выведите текст «нет символа»;
- если символ * встречается в строке только один раз, выведите его индекс;
- если символ * встречается два и более раз, выведите индекс его первого, второго и последнего вхождения, удалите первый и последний символ * из строки.

8. Дана строка, состоящая из слов, разделенных пробелами. В этой строке:
- удалите все лишние пробелы (в начале, в конце, между словами оставить ровно один пробел);
- поменяйте регистр символов (строчные сделать прописными, прописные – строчными);
- определите, сколько в ней слов.

9. Строка содержит фамилию, имя и отчество, записанные через пробелы. Например «Иванов Иван Иванович». Для этой строки:
- выведите информацию в виде:
```
Фамилия       Иванов
Имя           Иван
Отчество      Иванович
```
- получите строки вида «Иванов И.И.» и «И.И. Иванов».

10. Напишите программу, которая осуществляет вывод кошки на экран. Примечание: кошка выглядят примерно так:
```
         /\_/\ 
         /@ @\ 
        (~~0~~)

 ```

11. Для двух произвольных строк провести сравнение на совпадение содержимого без учета регистра букв и начальных и конечных пробельных символов. Например, для двух строк ' HeLLO WOrlD' и 'hello WORLD ' такое сравнение должно возвращать True (содержание строк совпадает).

12. По номеру года определите, является ли данный год високосным. (год является високосным, если его номер кратен 4, но не кратен 100, а также если он кратен 400).

18. Даны точки $A=(1,1),B=(5,5),C=(1,4),D=(7,2)$. Найти точку пересечения отрезков $AB$ и $CD$.

1.1 Значения переменных А и В ввести с клавиатуры и вывести на экран. После этого значения меняются местами, т.е. A нужно присвоить значение B, а B – значение A, и вновь значения переменных вывести на экран.

In [2]:
a = int (input("a ="))
b = int (input("b ="))
a, b = b, a
print ("a =", a)
print ("b =", b)

a =1
b =2
a = 2
b = 1


1.2 Значение x вводится с клавиатуры. Вычислите $y=(x+1)^2\cdot \sqrt{x^3+1}$ Выведите на экран значения $x$ и $y$ с тремя знаками после запятой.

In [1]:
x = int (input())
y = ((x+1)**2)*((x**3+1)**(1/2))
print ("x: ", x)
print ("y: ", round(y, 3))

1
x:  1
y:  5.657


1.3 Вычислите сумму цифр пятизначного числа.

In [5]:
s = int(input("Ввести пятизначное число: "))
a = s%10
b = (s//10)%10#пример использования операторов
c = ((s//10)//10)%10
d = (((s//10)//10)//10)%10
e = ((((s//10)//10)//10)//10)%10
print(f"{a}+{b}+{c}+{d}+{e}={a+b+c+d+e}")

Ввести пятизначное число: 15673
3+7+6+5+1=22


1.4 Для заданного трехзначного числа выведите число, у которого цифры идут в обратном порядке, например, для числа 123 ответ 321.

In [7]:
b = int(input("Ввести трёхзначное число: "))
b1 = b%10
b2 = (b//10)%10
b3 = ((b//10)//10)%10
print(b1, b2, b3, sep = '')

Ввести трёхзначное число: 670
076


1.5 Ввести координаты 2 точек: $(x_1,y_1)$ и $(x_2,y_2)$. Вычислите расстояние между этими точками.  Результат выведите с 5 знаками после запятой.

In [9]:
x1 = int(input("Ввести x1: "))
y1 = int(input("Ввести y1: "))
x2 = int(input("Ввести x2: "))
y2 = int(input("Ввести y2: "))
c = (((x2-x1)**2)+((y2-y1)**2))**(1/2)
print("Расстояние между точками", round(c,5))

Ввести x1: 1
Ввести y1: 8
Ввести x2: 0
Ввести y2: 2
Расстояние между точками 6.08276


1.6 Разработать программу вычисления по известному радиусу площади круга и длины окружности.

In [10]:
import math
c = int(input ("Ввести радиус: "))
v = math.pi*(c**2)
g = 2+math.pi*c
print("Площадь круга: ", round (v, 3), ", длина окружности: ", round(g, 3), sep = "")

Ввести радиус: 12
Площадь круга: 452.389, длина окружности: 39.699


1.7 Разработать программу по вычислению площади кольца по известным значениям его внешнего и внутреннего радиусов.

In [15]:
import math
r = int(input ("Ввести внутренний радиус: "))
R = int(input ("Ввести внешний радиус: "))
S = math.pi*((R**2)-(r**2))
print("Площадь кольца", round(S, 3))

Ввести внутренний радиус: 3
Ввести внешний радиус: 6
Площадь кольца 84.823


1.8 Разработать программу вычисления радиуса круга и его площади по известной длине окружности.

In [17]:
import math
c = int(input ("Ввести длину окружности: "))
r = c/(2*math.pi)
s = math.pi*(r**2)
print("Радиус равен: ", round(r, 3), ", площадь равна: ", round(s, 3), sep = "")

Ввести длину окружности: 2
Радиус равен: 0.318, площадь равна: 0.318


1.9 Разработать программу вычисления объема шара: а) по радиусу; б) по длине его экваториальной параллели. Объём шара рассчитывается по формуле $V=\frac{4}{3}\pi{r^3}$, где $r$ - радиус.

In [18]:
import math
print ("а) Расчёт по радиусу")
r = int(input("Ввести радиус: "))
g = (4/3)*math.pi*(r**3)
print("g =", round (g, 3))

а) Расчёт по радиусу
Ввести радиус: 3
g = 113.097


In [19]:
import math
print ("а) Расчёт по длине экваториальной параллели")
l = int(input("Ввести длину экваториальной параллели: "))
r = l/(2*math.pi)
g = (4/3)*math.pi*(r**3)
print("g =", round (g, 3))

а) Расчёт по длине экваториальной параллели
Ввести длину экваториальной параллели: 5
g = 2.111


1.10 Имеются два $n$–мерных вектора $x$ и $y$, которые задают координаты $n$ точек на плоскости (случайные целые числа). Найти наиболее близкие друг другу точки.

In [1]:
import random
n = 3
num_points = 5

x = []#создание пустого списка для хранения элементов вектора
for _ in range(num_points):#запускается цикл.
    point = [random.randint(1, 100) for _ in range(n)]#На каждой итерации функция random.randint(0,100) генерирует случайное число
    x.append(point)#Случайное число добавляется в конце списка vector с попомщью метода append

y = []
for _ in range(num_points):
    point = [random.randint(1, 100) for _ in range(n)]
    y.append(point)
def distance(point1, point2):
    squared_distance = sum((x - y) ** 2 for x, y in zip(point1, point2))
    return squared_distance ** 0.5

min_distance = None#инициализация переменной(присваивание начального значения,но неизвестно какое оно будет)
closest_points = None

for i in range(num_points):#Внешний цикл for i in range(num_points) перебирает числа от 0 до num_points - 1 и присваивает каждое значение переменной i.
    for j in range(num_points):#Внутренний цикл for j in range(num_points) похож на внешний цикл и также перебирает числа от 0 до num_points - 1, присваивая значение j.
        d = distance(x[i], y[j])
if min_distance is None or d < min_distance:#Выражение означает, что если min_distance равно None или d меньше min_distance, то выполняется условие
    min_distance = d
closest_points = (x[i], y[j])
print(f"Наиболее близкие точки: {closest_points[0]} и {closest_points[1]}")
print(f"Расстояние между ними: {min_distance:.2f}")

Наиболее близкие точки: [65, 81, 83] и [53, 93, 43]
Расстояние между ними: 43.45


1.11 Треугольник задан координатами своих вершин: $(x_1,y_1),(x_2,y_2),(x_3,y_3)$. Значения координат определите с помощью присваивания. Они могут быть нецелыми. Найти периметр и площадь треугольника.

In [20]:
x1, y1 = 2, 4
x2, y2 = 1, 6
x3, y3 = 3, 5
a = (((x2-x1)**2)+((y2-y1)**2))**(1/2)
b = (((x3-x2)**2)+((y3-y2)**2))**(1/2)
c = (((x3-x1)**2)+((y3-y1)**2))**(1/2)
g = a + b + c
gg = g/2
t = (gg*(gg-a)*(gg-b)*(gg-c))**(1/2)
print (a, b, c)
print ("Периметр треугольника: ", round(g, 3), ", площадь треугольника: ", round(t, 3), sep = "")

2.23606797749979 2.23606797749979 1.4142135623730951
Периметр треугольника: 5.886, площадь треугольника: 1.5


1.12 Пусть пользователь вводит 5 чисел: а, b, c, d, e. Реализуйте программу расчета выражения вида $res=\frac{a+bc}{2d-e}+Ost[b^e/c]$, где $Ost$ - остаток от деления. Учесть невозможность деления на 0.

In [22]:
import sys
a = int(input("Ввести значения a: "))
b = int(input("Ввести значения b: "))
c = int(input("Ввести значения c: "))
if c == 0:
    sys.exit("Некорректное значение переменной c. Деление на ноль")#выкидывает из программы
d = int(input("Ввести значения d: "))
e = int(input("Ввести значения e: "))
part1 = a+b+c
part2 = 2*d-e
if part2 == 0:
    sys.exit("Некорректное значения. Деление на ноль при вычислении")
res = (part1/part2) + ((b**e)%c)
print("Результат: ", res)

Ввести значения a: 2
Ввести значения b: 1
Ввести значения c: 5
Ввести значения d: 2
Ввести значения e: 3
Результат:  9.0


2.1 Введите строку, состоящую из двух цифр. Преобразуйте ее в целое и вещественное число. Выведите полученные три значения (строку, целое число, вещественное число) на экран в одной строке через запятую, затем пропустите строку и вновь выведите значения по одному на строке. Перед каждым значением выведите его тип.

In [23]:
s = input()
sint = int(s)
sfloat = float(s)
print(s, sint, sfloat, sep = ", ")
print(type(s), s) #type() - это встроенная функция, которая позволяет определить тип объекта
print(type(sint), sint)
print(type(sfloat), sfloat)

1
1, 1, 1.0
<class 'str'> 1
<class 'int'> 1
<class 'float'> 1.0


2.2 Для строки 'Финансовый университет' двумя способами получить подстроку с 1-го по 4-й символы включительно.

In [24]:
i = 0
d = "Финансовый университет"
d1 = " "
while i<4: #пока i меньше 4
    d1 = d1+d[i]
    i+=1
print(d1)

 Фина


In [25]:
d = "Финансовый университет"
print(d[0:4])

Фина


2.3 Составить строку из всех четных символов строки 'Финансовый университет'.

In [26]:
print("Финансовый университет"[::2])

Фнноы нврие


2.4 Получить подстроку неизвестной заранее строки, содержащую половину символов строки и расположенную посередине строки.

In [27]:
r = input()
b = len(r)
h = (b/2)/2
h = int(h)
print(r[h:-h])

Привет
риве


2.5 Коротко записать создание строки 'oneoneoneoneoneonetwotwotwo'

In [28]:
print('one'*6, 'two'*3, sep = "")

oneoneoneoneoneonetwotwotwo


2.6 Для введенной строки выведите (на отдельной строке):
- второй символ этой строки;
- предпоследний символ этой строки;
- всю строку, кроме последних двух символов;
- все символы с четными индексами (считая, что индексация начинается с 0, поэтому символы выводятся, начиная с первого);
- все символы с нечетными индексами, то есть начиная со второго символа строки;
- все символы в обратном порядке;
- все символы строки через один в обратном порядке, начиная с последнего;
- длину данной строки.

In [2]:
x = input()
print("Второй символ строки: ", x[1])
n = len(x)
print("Предпоследний символ строки: ", x[n-2])
print("Вся строка, кроме послдених двух символов: ", x[:(n-2)])
print("Все символы с чётными индексами: ", x[::2])
print("Все символы с нечётными индексами: ", x[1::2])
print("Все в обратном порядке: ", x[::-1])
print("все символы строки через один в обратном порядке, начиная с последнего: ", x[n::-2])
print("Длина строки: ", n)

bubble
Второй символ строки:  u
Предпоследний символ строки:  l
Вся строка, кроме послдених двух символов:  bubb
Все символы с чётными индексами:  bbl
Все символы с нечётными индексами:  ube
Все в обратном порядке:  elbbub
все символы строки через один в обратном порядке, начиная с последнего:  ebu
Длина строки:  6


2.7 Не используя метод count, для заданной строки выполните:
- если символ * в данной строке отсутствует, выведите текст «нет символа»;
- если символ * встречается в строке только один раз, выведите его индекс;
- если символ * встречается два и более раз, выведите индекс его первого, второго и последнего вхождения, удалите первый и последний символ * из строки.

In [2]:
s = "Actions"
a = s.find('i')#метод выполняет поиск подстроки 'i' в строке s и возвращает индекс (позицию) первого вхождения этой подстроки в строку s.
b = s.rfind('i')#метод возвращает наивысший индекс подстроки (если найден)
if a == -1:
    print("Нет символа")
if a == b:
    print(a+1)
else:
    print(a, b)
    print(s)

4


2.8 Дана строка, состоящая из слов, разделенных пробелами. В этой строке:
- удалите все лишние пробелы (в начале, в конце, между словами оставить ровно один пробел);
- поменяйте регистр символов (строчные сделать прописными, прописные – строчными);
- определите, сколько в ней слов.

In [8]:
s =(' '.join(input("Ввести строку: ").split()))
print (s)
print (s.swapcase())#меняет регистры
print (s.count(" ")+1)#определяет сколько в ней слов

Ввести строку: Tornado
Tornado
tORNADO
1


2.9 Строка содержит фамилию, имя и отчество, записанные через пробелы. Например «Иванов Иван Иванович». Для этой строки:
- выведите информацию в виде:
```
Фамилия       Иванов
Имя           Иван
Отчество      Иванович
```
- получите строки вида «Иванов И.И.» и «И.И. Иванов».

In [2]:
s1 = "Иванов Иван Ивановыч"
s2 = "Фамилия Имя Отчество"
print(s2[:s2.find(" ")], "  ",s1[:s1.find(" ")])
print(s2[s2.find(" ")+1:s2.rfind(" ")], "  ",s1[s1.find(" "):s1.rfind(" ")] )
print(s2[s2.rfind(" ")+1:], "  ",s1[s1.rfind(" "):])

Фамилия    Иванов
Имя     Иван
Отчество     Ивановыч


In [8]:
lastname = "Иванов Иван Ивановыч".split(' ', 1)[-2]#отделяет Иванов и Иван Иванович, а потом с помощью индекса[-2] отчитывает с конца и выводит Иванов
name = "Иванов Иван Иванович".split(' ', -1)[1]
patronymic = "Иванов Иван Иванович".split(' ', 2)[-1]
print(lastname, " ", name[0], '.', patronymic[0],'.', sep = '')

Иванов И.И.


In [7]:
lastname = "Иванов Иван Ивановыч".split(' ', 1)[-2]
name = "Иванов Иван Иванович".split(' ', -1)[1]
patronymic = "Иванов Иван Иванович".split(' ', 2)[-1]
print(name[0], '.', patronymic[0], '.', " ", lastname, sep = '')

И.И. Иванов


2.10 Напишите программу, которая осуществляет вывод кошки на экран. Примечание: кошка выглядят примерно так:
```
         /\_/\ 
         /@ @\ 
        (~~0~~)

 ```


In [24]:
print("   /\_/\ \n   /@ @\ \n  (~~0~~)")

   /\_/\ 
   /@ @\ 
  (~~0~~)


2.11 Для двух произвольных строк провести сравнение на совпадение содержимого без учета регистра букв и начальных и конечных пробельных символов. Например, для двух строк ' HeLLO WOrlD' и 'hello WORLD ' такое сравнение должно возвращать True (содержание строк совпадает).

In [9]:
s1 = "cocos"
s2 = "coCos"
s3 = "cocas"
print(s1.lower()==s2.lower())#lower()-делает все буквы прописными
print(s1.lower()==s3.lower())

True
False


2.12 По номеру года определите, является ли данный год високосным. (год является високосным, если его номер кратен 4, но не кратен 100, а также если он кратен 400).

In [10]:
year = int (input ())
if year%4==0:
    if year%100!=0:
        if year%400==0:
            print ("Високосный год")
else:
    print ("Не високосный год")

2002
Не високосный год


2.13 Даны точки $A=(1,1),B=(5,5),C=(1,4),D=(7,2)$. Найти точку пересечения отрезков $AB$ и $CD$.

In [2]:
x1,y1=(5,15)
x2,y2=(5,0)
x3,y3=(0,10)
x4,y4=(10,10)
b1=(x1*y2-y1*x2)*(x3-x4)-(x1-x2)*(x3*y4-y3*x4)
k1=(x1-x2)*(y3-y4)-(y1-y2)*(x3-x4)
b2=(x1*y2-y1*x2)*(y3-y4)-(y1-y2)*(x3*y4-y3*x4)
k2=(x1-x2)*(y3-y4)-(y1-y2)*(x3-x4)
x_interset=b1/k1
y_interset=b2/k2
print("Точка пересечения отрезков:",(x_interset,y_interset))

Точка пересечения отрезков: (5.0, 10.0)
